lib/engine/step/dividend.rb

```diff
@@ -28,9 +28,19 @@ def skip!
         process_dividend(Action::Dividend.new(current_entity, kind: 'withhold'))
       end
 
+      def dividend_options(entity)
+        revenue = routes.sum(&:revenue)
+        dividend_types.map do |type|
+          payout = send(type, entity, revenue)
+          [type, payout.merge(share_price_change(entity, revenue - payout[:company]))]
+        end.to_h
+      end
+
       def process_dividend(action)
         entity = action.entity
         revenue = routes.sum(&:revenue)
+        payout = dividend_options(entity)[action.kind.to_sym]
```

Comment:
should we instead change it to string? i see on line 56 you don't use to_sym, could be danagerous?

Reply:
When coming from a load it's a string, when coming from the UI it's a sym. I've converted it to sym uniformly.